packages/core/src/app/checkout/AnalyticsEvents.ts

```diff
@@ -21,34 +19,79 @@ export enum GuestCheckoutEvents {
   Exit = "Exit"
 }
 
+const defaultTimeout = 1000;
+const defaultPollingInterval = 10;
+// borrowed polling mechanism 
+async function pollUntilAccepted<T>(
+  fn: () => T,
+  accept: (t: T) => boolean,
+  timeoutMillis = defaultTimeout,
+  interval = defaultPollingInterval,
+): Promise<T | Error> {
+  return new Promise<T | Error>(resolve => {
```

Comment:
Is there a reason to return a new promise when this is an async function which already returns a promise?

Reply:
yeah good call out, this is redundant for sure